src/graph.rs

```diff
@@ -39,6 +43,62 @@ impl<'a, N: Clone, E: Clone, Ty: EdgeType> Graph<N, E, Ty> {
         })
     }
 
+    pub fn edge_by_screen_pos(
+        &self,
+        meta: &'a Metadata,
+        style: &'a SettingsStyle,
+        screen_pos: Pos2,
+    ) -> Option<(EdgeIndex, &Edge<E>)> {
+        let pos_in_graph = (screen_pos.to_vec2() - meta.pan) / meta.zoom;
+        self.edges_iter().find(|(idx, e)| {
+            let (source, target) = self.g.edge_endpoints(*idx).unwrap();
+            let source_pos = self.g.index(source).location();
+            let target_pos = self.g.index(target).location();
+            if source == target { // edge is a loop (bezier curve)
+                // TODO lots of duplicated code (only slightly changed from draw/edge
+                let node = self.g.index(source);
+                let rad = node.screen_radius(meta, style) / meta.zoom;
+                let center = source_pos;
+                let center_horizon_angle = PI / 4.;
+                let y_intersect = center.y - rad * center_horizon_angle.sin();
+
+                let edge_start = Pos2::new(center.x - rad * center_horizon_angle.cos(), y_intersect);
+                let edge_end = Pos2::new(center.x + rad * center_horizon_angle.cos(), y_intersect);
+
+                // TODO what does `order` do?
+                let loop_size = rad * style.edge_looped_size;
+
+                let control_point1 = Pos2::new(center.x + loop_size, center.y - loop_size);
+                let control_point2 = Pos2::new(center.x - loop_size, center.y - loop_size);
+
+                let stroke = Stroke::new(e.width() * meta.zoom, e.color(&Default::default()));
+                let shape = CubicBezierShape::from_points_stroke(
+                    [edge_end, control_point1, control_point2, edge_start],
+                    false,
+                    Color32::TRANSPARENT,
+                    stroke,
+                );
+                let tolerance = Option::new(1.0);
+                let mut previous_point = None;
+                for p in shape.flatten(tolerance) {
+                    if previous_point.is_some() {
+                        let distance = distanceSegmentToPoint(p.to_vec2(), previous_point.unwrap(), pos_in_graph);
+                        if distance < e.width() {
+                            return true;
+                        }
+                    }
+                    previous_point = Option::new(p.to_vec2());
+                };
+                return false;
+            } else { // edge is a straight line between nodes
```

Comment:
Order should be used in this clause to select proper edge from the set of duplicated edges.